z3/src/z3_term.cpp

```diff
@@ -104,26 +104,159 @@ bool Z3Term::compare(const Term & absterm) const
 
 Op Z3Term::get_op() const
 {
-  throw NotImplementedException("get_op not yet implemented.");
+  if (is_function)
+  {
+    throw IncorrectUsageException("A function has no operator.");
+  }
+  else
+  {
+    if (!term.is_app())
+    {
+      throw IncorrectUsageException(
```

Comment:
This should also return a null operator I think (`get_op` shouldn't throw exceptions). What kind of terms have false for `is_app()`?